lib/ansiblelint/formatters/__init__.py

```diff
@@ -41,69 +43,72 @@ def _format_path(self, path):
 
 class Formatter(BaseFormatter):
 
-    def format(self, match, colored=False):
+    def format(self, match: MatchError, colored: bool = False) -> str:
         formatstr = u"{0} {1}\n{2}:{3}\n{4}\n"
         _id = getattr(match.rule, 'id', '000')
         if colored:
-            return formatstr.format(colorize(u"[{0}]".format(_id), Color.error_code),
-                                    colorize(match.message, Color.error_title),
-                                    colorize(self._format_path(match.filename), Color.filename),
-                                    colorize(str(match.linenumber), Color.linenumber),
-                                    colorize(u"{0}".format(match.details), Color.line))
+            return formatstr.format(
+                colorize(u"[{0}]".format(_id), Color.error_code),
+                colorize(match.message, Color.error_title),
+                colorize(self._format_path(match.filename or ""), Color.filename),
+                colorize(str(match.linenumber), Color.linenumber),
+                colorize(u"{0}".format(match.details), Color.line))
```

Comment:
looks unrelated

Reply:
**** it is related! If you would try to add some typing information you will see it. Typing complains about not allowing None, so you add `or ""` part and the line becomes too long, so you need to wrap it to pass the linters job.

Also, colorize requires color Literals, not strings but we forgot to update some calls when we did it. Now mypy does slap us, forcing to fix the bug.

You will see more of these. That is why I said is better to add typing info in chunks that keep the review size decent (ideally under 50 lines).